dbos-rules.js

```diff
@@ -53,9 +54,35 @@ const extConfig =
 module.exports = {
   meta: {
     "name": "@dbos-inc/eslint-plugin",
-    "version": "0.0.2",
+    "version": "0.0.4",
   },
   rules: {
+    'detect-native-code': {
+      // Rule configuration for detection of libraries based on native code
+      meta: {
+        type: 'suggestion',
+        docs: {
+          description: 'Detect calls to libraries with native functions like bcrypt, which should be replaced with native JS',
+        },
+        schema: [],
+      },
+      create: function (context) {
+        return {
+          CallExpression(node) {
+            //console.log(node.callee.type+JSON.stringify(node));
+            if (node.callee.type === 'MemberExpression' &&
+                node.callee.object.name === 'bcrypt' &&
+                (node.callee.property.name === 'compare' || node.callee.property.name === 'hash'))
```

Comment:
I'm not sure how eslint work, but I thought it would be sufficient to check import/require statements? (rather than looking them up the entire AST). My experience with eslint is that it gets pretty slow on large codebases. Maybe that's something we can keep in mind and handle in the future?

Reply:
Also looking at import statements would make sense.